### PR TITLE
Add Elasticsearch 7 support to regulations search behind a flag

### PR DESCRIPTION
**Pull request by alexm118**
Places ES7 search functionality behind a toggle.

---

## Additions

- Moving search code to work with ES7 behind flag.
- Add short_name to index to reduce need to query DB for data needed.
- configure boost for text field to improve results
- swap section_order to Keyword Field so we can sort by it.

## How to test this PR

1. Enable `ELASTICSEARCH_DSL_REGULATIONS` flag
2. Load up ES7 index `cfgov/manage.py search_index --rebuild -f --parallel`
3. Search regulations

## Notes and todos

- Seeing a large discrepancy in search result counts. "mortgage" is off by nearly 300 records.

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

**Review by Scotchester — APPROVED** [review]
Works for me! Both the new ES7 indexing/searching with the flag, and using ES2 without the flag.

File: cfgov/regulations3k/models/pages.py [hunk 1]
```diff
@@ -56,8 +58,94 @@ def get_template(self, request):
...
+        })
+        return TemplateResponse(
+            request,
+            self.get_template(request),
+            context)
+
     @route(r'^results/')
     def regulation_results_page(self, request):
+        if flag_enabled('ELASTICSEARCH_DSL_REGULATIONS'):
+            return self.regulation_results_page_es7(request)
```
Comment (willbarton) [inline cfgov/regulations3k/models/pages.py:148]: This is fine, but it makes me wonder if Wagtail-Flags should include a `flagged_route()` decorator for routable pages.

**Review by willbarton — APPROVED** [review]

File: cfgov/regulations3k/models/pages.py (hunk 1 above)
Comment (alexm118) [reply cfgov/regulations3k/models/pages.py:148]: Definitely seems like something that could be valuable, especially outside these niches with things like A/B Testing pages

**Review by higs4281 — APPROVED** [review]
This appears to be fixing a giant hole in our current search setup.

This branch returns result sets that are 1 to 8 times bigger than results on www.

In addition to finding case-insensitive matches that are being missed, our current search also is simply missing direct hits, such as this one:

Search term: "errors"
Reg: E (1005)

The graf [`1005/12/#b-2-iii`](https://www.consumerfinance.gov/policy-compliance/rulemaking/regulations/1005/12/#b-2-iii) contains the word "errors" but doesn't show up in [live results](https://www.consumerfinance.gov/policy-compliance/rulemaking/regulations/search-regulations/results/?q=errors&regs=1005&results=50&order=relevance).  
In our ES7 branch, it shows up.

Can't wait to get this running. Bravo!

**Pull request merged**